platforms/documentation/docs/src/docs/userguide/optimizing-performance/configuration_cache.adoc

```diff
@@ -989,8 +989,10 @@ Use <<dataflow_actions.adoc#dataflow_action,dataflow actions>> to handle the bui
 
 Plugin and build scripts should avoid running external processes at configuration time.
 In general, it is preferred to run external processes in tasks with properly declared inputs and outputs to avoid unnecessary work when the task is up-to-date.
-If necessary, only configuration-cache-compatible APIs should be used instead of Java and Groovy standard APIs or existing `ExecOperations`,
-`Project.exec`, `Project.javaexec`, and their likes in settings and init scripts.
+However, should the need arise, only configuration-cache-compatible APIs described below must be used.
+Using Java and Groovy standard APIs or existing `ExecOperations`, `Project.exec`, `Project.javaexec`, and their likes in settings and init scripts is not allowed when the configuration cache is enabled.
```

Comment:
Why do we single out settings and init scripts (and omit build scripts) when saying those features not allowed - if they are not allowed in build scripts either?

Reply:
> Why do we single out settings and init scripts (and omit build scripts) when saying those features not allowed - if they are not allowed in build scripts (at configuration time) either?

I don't think this is intentional, fixed. Nice catch!